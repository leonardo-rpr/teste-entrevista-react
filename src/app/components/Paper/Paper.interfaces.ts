import React from 'react';

export interface PaperInterface {
  title?: string | null;
  children?: React.ReactNode | null;
  width?: string;
}

export interface PaperStylesInterface {
  hasChildren?: boolean;
  hasTitle?: boolean;
  width?: string;
}
