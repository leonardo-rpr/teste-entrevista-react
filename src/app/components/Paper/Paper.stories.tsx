import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import Paper from './Paper';

const stories = storiesOf('Components', module);

stories.add('Paper', () => {
  const children = text('children', 'Hello world!');
  const title = text('title', 'Lorem ipsum dolor sit amet');

  return <Paper title={title}>{children}</Paper>;
});
