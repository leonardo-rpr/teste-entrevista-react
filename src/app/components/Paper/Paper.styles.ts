import styled, { css } from 'styled-components';
import { PaperStylesInterface } from './Paper.interfaces';

export const Container = styled.div<PaperStylesInterface>`
  background: #fff;
  border-radius: 12px;
  display: inline-block;
  width: 100%;
  transition: all 0.25s ease-in;
  font-weight: 300;
  font-size: 12px;
  color: #565656;
  box-shadow: 0px 10px 14px 1px rgba(60, 64, 67, 0.08), 0px 6px 6px -3px rgba(60, 64, 67, 0.16);

  :hover {
    box-shadow: 0px 14px 21px 2px rgba(60, 64, 67, 0.08), 0px 6px 7px -4px rgba(60, 64, 67, 0.16);
  }
`;

export const Wrapper = styled.div<PaperStylesInterface>`
  margin: 15px;
`;

export const Title = styled.h2<PaperStylesInterface>`
  font-size: 18px;
  font-weight: 600;
  color: #aaaaaa;
  padding: 20px;
  display: block;
  ${({ hasChildren }) => css`
    ${hasChildren && 'padding-bottom: 10px;'}
  `}
`;

export const Children = styled.div<PaperStylesInterface>`
  padding: 20px;
  ${({ hasTitle }) => css`
    ${hasTitle && 'padding-top: 10px;'}
  `}
`;
