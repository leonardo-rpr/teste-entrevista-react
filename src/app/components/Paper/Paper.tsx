import React from 'react';
import { Wrapper, Container, Children, Title } from './Paper.styles';

import { PaperInterface } from './Paper.interfaces';

const Paper: React.FC<PaperInterface> = ({ title, children }) => (
  <Wrapper>
    <Container>
      {title && <Title hasChildren={!!children}>{title}</Title>}
      {children && <Children hasTitle={!!title}>{children}</Children>}
    </Container>
  </Wrapper>
);

Paper.defaultProps = {
  children: null,
  title: null,
};

export default Paper;
