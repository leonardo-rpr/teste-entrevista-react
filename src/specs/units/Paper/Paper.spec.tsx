import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Paper from '../../../app/components/Paper';

afterEach(cleanup);

describe('Paper', () => {
  it('Renders at the default state correctly', () => {
    const wrapper = render(<Paper />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Renders at the default state with title correctly', () => {
    const wrapper = render(<Paper title="Lorem ipsum dolor sit amet" />);

    expect(wrapper).toMatchSnapshot();
  });
});
