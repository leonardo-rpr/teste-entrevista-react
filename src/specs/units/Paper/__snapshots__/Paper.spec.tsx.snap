// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper Renders at the default state correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa cyJbOz"
        width="100%"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa cyJbOz"
      width="100%"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Paper Renders at the default state with title correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa cyJbOz"
        width="100%"
      >
        <h2
          class="sc-bwzfXH eYMHTJ"
        >
          Lorem ipsum dolor sit amet
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa cyJbOz"
      width="100%"
    >
      <h2
        class="sc-bwzfXH eYMHTJ"
      >
        Lorem ipsum dolor sit amet
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
